### DISCUSSION TITLE
Mingling Korean Doc Image with Korean Text

### DISCUSSION THREAD

**Discussion by cweiler-blumatix** [pymupdf/PyMuPDF#2372] [Looking for help] [answered]
Hi,

I got following issue: My current OCR does not support Korean. But it seems it supports text extraction from a text-pdf with Korean characters. Thus, I thought I might just use another OCR for Korean and create a text-pdf which my existing OCR will pick up.

Thus, I managed to:
- Use easy ocr to retrieve Korean text out of an doc (png image)
- Created with PyMuPDF a new PDF including the text
 ```
# Setup new page
page_pdf = doc.new_page(width=image.width, height=image.height)
page_rect = fitz.Rect(0, 0, image.width, image.height)
page_pdf.insert_image(page_rect, pixmap=image)
...
#font
font = fitz.Font(insert_text_font)
if insert_text_font == "cjk":
      page_pdf.insert_font(fontname="F0", fontbuffer=font.buffer)
      insert_text_font="F0"
...
#insert text
font_size_opt = select_font_size(bbox['Height'], bbox['Width'], text, font)
res = page_pdf.insert_textbox(rect, text, fontsize=font_size_opt, fontname=insert_text_font, align=1, stroke_opacity=instert_text_opacity, fill_opacity=instert_text_opacity)
```
-  Now  I can copy and paste the text e.g. into a text editor out of PDF Acrobat Reader View successfully

However, my OCR does not pick it up? Do I miss something substantially?
I understand that for CJK a font is embedded - but somehow it seems my existing OCR does not pick up the embedded text at all?
Any help or idea is welcome. 

Thanks

**Comment by JorjMcKie** [comment]
What is your "current OCR"? easyocr? Why don't you try PyMuPDF's builtin interface to Tesseract-OCR (which must be installed of course). When invoking it, make sure to supply the language spec for Korean to it.
That too might be the problem of your current OCR engine.

**Comment by JorjMcKie** [comment, marked as answer]
Ah sorry,did not fully understand your comment in the first place:
No: an OCR engine only interprets images. Standard text therefore is not taken into account.
In PyMuPDF, there are ways to deal with a micture of OCRed and standard text.

**Reply by cweiler-blumatix** [reply]
Thanks - we use Nuance. And it can extract text layer from pdf and uses it. But not in this case. I see that it runs into an exception since it tries to create images that are huge. Maybe related to the embedded font. Not sure - but nothing for this discussion. Anyway thanks for your quick reply.
